### PR TITLE
[Story Player] Expose goToPageId functionality

### PR DESCRIPTION
**Pull request by mszylkowski**
Follows https://github.com/ampproject/amphtml/pull/30927

Closes https://github.com/ampproject/amphtml/issues/31047

Programmatically sends the player to the page specified by the ID passed to the method `goToPageId`

### CODE REVIEW

File: src/amp-story-player/amp-story-player-impl.js [hunk 1]
```diff
@@ -1271,6 +1271,17 @@ export class AmpStoryPlayer {
     });
   }
 
+  /**
+   * @param {string} pageId
+   * @public
+   */
+  goToPageId(pageId) {
```
Comment (Enriqe) [inline src/amp-story-player/amp-story-player-impl.js:1278]: We might want to join this together with the `show(story, pageId)` API, this way we can control the code flow whenever a publisher wants to switch a story AND go to a specific page. Otherwise it would be an easy way to get a race condition.

File: src/amp-story-player/amp-story-player-impl.js [hunk 2]
```diff
@@ -789,6 +795,10 @@ export class AmpStoryPlayer {
 
     this.currentIdx_ = storyIdx;
     this.onNavigation_();
+
+    if (pageId != null) {
```
Comment (Enriqe) [inline src/amp-story-player/amp-story-player-impl.js:799]: Having two calls to `goToPageId_()` in two different spots will make this code hard to follow. Can we wrap the logic to switch to a different story to another method and put that inside a conditional instead?

File: src/amp-story-player/amp-story-player-impl.js [hunk 3]
```diff
@@ -736,59 +736,64 @@ export class AmpStoryPlayer {
   }
 
   /**
-   * Shows the story provided by the URL in the player.
+   * Shows the story provided by the URL in the player and go to the page if provided.
    * @param {string} storyUrl
```
Comment (Enriqe) [inline src/amp-story-player/amp-story-player-impl.js:740]: nit: `!string` if we want to accept null.

**Review by Enriqe — APPROVED** [review]
LGTM! 🎉

File: src/amp-story-player/amp-story-player-impl.js (hunk 3 above)
Comment (gmajoulet) [reply src/amp-story-player/amp-story-player-impl.js]: Well no that'd be `?string` 😅

Comment (mszylkowski) [reply src/amp-story-player/amp-story-player-impl.js]: Haha good catch
